From: =?utf-8?b?Ik1hcmNvIFRyZXZpc2FuIChUcmV2acOxbyki?= <mail@3v1n0.net>
Date: Wed, 15 Aug 2018 03:33:13 +0200
Subject: js/main: Throw error if no valid default stylesheet is found

In case a mode uses a stylesheet that can't be loaded, throw an error using an
informative error, instead of crashing later because invalid path is set

Forwarded: not-needed, upstream does not support custom modes/themes officially.
---
 js/ui/main.js | 3 +++
 1 file changed, 3 insertions(+)

diff --git a/js/ui/main.js b/js/ui/main.js
index aa1de83..92e61c8 100644
--- a/js/ui/main.js
+++ b/js/ui/main.js
@@ -346,6 +346,9 @@ function loadTheme() {
     let theme = new St.Theme ({ application_stylesheet: _cssStylesheet,
                                 default_stylesheet: _defaultCssStylesheet });
 
+    if (theme.default_stylesheet == null)
+        throw new Error("No valid stylesheet found for '%s'".format(sessionMode.stylesheetName));
+
     if (previousTheme) {
         let customStylesheets = previousTheme.get_custom_stylesheets();
 
